### DISCUSSION TITLE
Data Model-View-Reducer naming convention

### DISCUSSION THREAD

**Discussion by andrewjroberts** [pointfreeco/swift-composable-architecture#1335] [Beta]
Hi TCA community,

What is the naming convention for data model-view-reducer triplets?

**For example**,
Story <-- Data Model
StoryView <-- View
Story___ <-- ReducerProtocol conforming struct

Should the ___ be:
- Reducer
- Core
- Store
- A verb related to the feature, e.g. StoryReader

And then, would the rule be summarized as:
Prefer nothing but use {INSERT ____} in case of naming conflict?

---

@ contributors, ReducerProtocol is the bees knees, I'm having so much fun using it.

**Comment by stephencelis** [comment]
@andwrobs We think different folks will come up with different conventions, but in the repos most of our features follow this convention:

```swift
struct Story: ReducerProtocol {
  struct State { /* ... */ } // <- Data model
  enum Action { /* ... */ }
  // ...
}

struct StoryView: View {
  let store: StoreOf<Story>
  // ...
}
```

**Reply by andrewjroberts** [reply]
thank you for responding!
